crowbar_framework/app/models/node_object.rb

```diff
@@ -1349,6 +1349,22 @@ def link_to_device?(device, linkname)
     # device is i.e. "/dev/sda", "/dev/vda", ...
     # linkname is i.e. "/dev/disk/by-path/pci-0000:00:04.0-virtio-pci-virtio1"
     return true if device == linkname
+
+    # If this looks like a virtio disk and the target platform is one
+    # that might not have the "by-path" links (e.g. SLES 12). Avoid
+    # using "by-path".
+    if linkname =~ /^\/dev\/vd[a-z]+$/
+      virtio_by_path_platforms = %w(
+        ubuntu-12.04
+        redhat-6.2
+        redhat-6.4
+        centos-6.2
+        centos-6.4
+        suse-11.3
+      )
+      return false unless virtio_by_path_platforms.include?(@node[:target_platform])
+    end
```

Comment:
> @rhafer Thank you, I moved it to link_to_device

Hm, I guess you missunderstood me a bit. I think this is not the correct fix. IMO the correct fix make sure that ```linked_devs``` cannot be nil a few lines below. (Though I really wonder how that can still happen).